### PR TITLE
Implements `ipnfo tool is_v4in6`

### PR DESCRIPTION
**Pull request by Harisabdullah**

### CODE REVIEW

File: ipinfo/cmd_tool.go [hunk 1]
```diff
@@ -33,6 +34,7 @@ Commands:
   aggregate    aggregate IPs, IP ranges, and CIDRs.
   lower        get start IP of IPs, IP ranges, and CIDRs.
   upper        get end IP of IPs, IP ranges, and CIDRs.
+  is4In6       reports whether ip is an IPv4-mapped IPv6 address.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:37]:
```suggestion
  is4In6       get whether the IP is an IPv4-mapped IPv6 address.
```

File: ipinfo/cmd_tool_is4In6.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	},
+}
+
+// printHelpToolIs4In6 prints the help message for the "is4In6" command.
+func printHelpToolIs4In6() {
+	fmt.Printf(
+		`Usage: %s tool is4In6 [<opts>] <ips>
+
+Description:
+  Reports whether given ip is an IPv4-mapped IPv6 address.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is4In6.go:25]: Similar as above

File: ipinfo/cmd_tool_is4In6.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Examples:
+  %[1]s is4In6 "::7f00:1"
+  %[1]s is4In6 "::ffff:
+  %[1]s is4In6 "::ffff:
+Options:
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is4In6.go:31]: Need newline above

File: ipinfo/cmd_tool_is4In6.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	fmt.Printf(
+		`Usage: %s tool is4In6 [<opts>] <ips>
+
+Description:
+  Reports whether given ip is an IPv4-mapped IPv6 address.
+
+Examples:
+  %[1]s is4In6 "::7f00:1"
+  %[1]s is4In6 "::ffff:
+  %[1]s is4In6 "::ffff:
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is4In6.go:30]: Do these all return `true` or `false` or a mix? Need a mix

File: lib/cmd_tool_is4In6.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+func CmdToolIs4In6(f CmdToolIs4In6Flags, args []string, printHelp func()) error {
+	if f.Help {
+		printHelp()
+		return nil
+	}
+
+	op := func(input string, inputType INPUT_TYPE) error {
+		switch inputType {
+		case INPUT_TYPE_IP:
+			fmt.Printf("%s %t\n", input, is4in6(input))
```
Comment (UmanShahzad) [inline lib/cmd_tool_is4In6.go:32]:
```suggestion
			fmt.Printf("%s,%t\n", input, is4in6(input))
```

File: lib/cmd_tool_is4In6.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+		default:
+			return ErrInvalidInput
+		}
+		return nil
+	}
+
+	return GetInputFrom(args, true, true, op)
+}
+
+func is4in6(ip string) bool {
```
Comment (UmanShahzad) [inline lib/cmd_tool_is4In6.go:42]: Is there no func for this within the netip library we intended on using for all of these?

**Review by UmanShahzad — COMMENTED** [review]
Needs to follow subcmd naming style as used in others (snake case not pascal / camel case).

Also see original linear issue to see intended spelling of the command.

File: lib/cmd_tool_is4In6.go (hunk 6 above)
Comment (Harisabdullah) [reply lib/cmd_tool_is4In6.go:42]: yes. net library doesn't have a function implementing this. It is defined in netIP library ( which we are using as inspiration ) but its not imported

**Comment by UmanShahzad** [comment]
> yes. net library doesn't have a function implementing this. It is defined in netIP library ( which we are using as inspiration ) but its not imported

Ah, I intended to use it in practice, not just for inspiration @Harisabdullah ! Same is for all the other new tools.

**Review by UmanShahzad — COMMENTED** [review]
See https://github.com/ipinfo/cli/pull/168#issuecomment-1699560476

File: ipinfo/cmd_tool_isV4In6.go [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	},
+}
+
+// printHelpToolIsV4In6 prints the help message for the "is_v4in6" command.
+func printHelpToolIsV4In6() {
+	fmt.Printf(
+		`Usage: %s tool is_v4in6 [<opts>] <ips>
+
+Description:
+  Reports whether given ip is an IPv4-mapped IPv6 address.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_isV4In6.go:25]: Should match short desc above

Comment (UmanShahzad) [reply ipinfo/cmd_tool_isV4In6.go:25]:
```suggestion
  get whether the IP is an IPv4-mapped IPv6 address.
```

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**